notebooks/11-xarray-intro.md

```diff
@@ -1,185 +1,68 @@
-
-```{code-cell} ipython3
-import numpy as np
-```
-
-We could also read the GeoTiff data directly into a numpy array using `rasterio`:
-
-```{code-cell} ipython3
-import rasterio
-
-with rasterio.open(data_file) as src:
-    herstappe_array = src.read()
-    herstappe_meta = src.meta
-```
-
-```{code-cell} ipython3
-herstappe_array
-```
-
-As we learnt in the previous lesson, Rasterio returns a Numpy `ndarray`:
-
-```{code-cell} ipython3
-type(herstappe_array)
-```
-
-Numpy supports different `dtype`s (`float`, `int`,...), but all elements of an array do have the same dtype. Note that NumPy auto-detects the data-type from the input.
-
-```{code-cell} ipython3
-herstappe_array.dtype
-```
-
-The data type of this specific array `herstappe_array` is float32. More information on the data types Numpy supports is available in the [documentation](https://numpy.org/devdocs/user/basics.types.html#array-types-and-conversions-between-types).
-
-+++
-
-Converting to another data type is supported by `astype` method. When floats are preferred during calculation:
-
-```{code-cell} ipython3
-herstappe_array.astype(int)
-```
-
-Just as any other object in Python, the `ndarray` has a number of attributes. We already checkes the `dtype` attribute. The `shape` and `ndim` of the array are other relevant attribute:
-
-```{code-cell} ipython3
-herstappe_array.shape, herstappe_array.ndim
-```
-
-Hence, we have three bands with dimensions (227, 447) and data type `float32`. Compare this to the metadata stored in the geotiff file:
-
-```{code-cell} ipython3
-# !gdalinfo ./data/herstappe/raster/2020-09-17_Sentinel_2_L1C_True_color.tiff
+ax = herstappe.plot.imshow(figsize=(12, 5))
+ax.axes.set_aspect('equal')
 ```
 
-The metadata on the dimensions and the datatype correspond, but the spatial information is lost when we only store the Numpy array. While if we use `xarray.DataArray` to wrap this NumPy array, we can keep the data and the spatial context together.
-
-+++
-
-<div class="alert alert-info" style="font-size:120%">
-
-**REMEMBER**: <br>
-
-The [`numpy` package](https://numpy.org/) is the backbone of the scientific Python ecosystem. The `ndarray` provides an efficient data type to store and manipulate raster data, but it does NOT contain any spatial information.
-
-The [11-numpy.ipynb](11-numpy.ipynb) notebook provides more content on using pure numpy with rasterio and how to keep track of the spatial extent (eg when plotting), but for this notebooks, we will use xarray to provide context to our numpy array dimensions.
-
-</div>
-
-+++
-
-Let's take a look at our xarray.DataArray again:
+We can further make the dimension information more useful by renaming the labels of the 'band' dimension:
 
 ```{code-cell} ipython3
+herstappe = herstappe.assign_coords(band=("band", ["red", "green", "blue"]))
 herstappe
 ```
 
-We can further make the dimension information more useful by renaming the labels of the "band" dimension:
+Or just rename the coordinate names itself:
```

Comment:
```suggestion
Or just rename the dimension names itself:
```

Or not .. In the overview list we in the previous notebook (it might be worth repeating that in this notebook) we call it dimensions. 

I think typically xarray "dimensions" for the names of the dimensions, and "coordinates" for the actual coordinate values for a specific dimension (which is of course not a clear cut line).

Reply:
I would keep coordinate here, the docstring of `rename` says also coordinate

> Returns a new DataArray with renamed coordinates or a new name.

I see these are 'dimension coordinates' (whereas you can also have non-dimension coordinates), so these are the names of the coordinates linked to the dimension. A dimension does not have a name on itself... _I'm realizing this while typing, so not sure if I did this consistently in the course notes_.

Reply:
> A dimension does not have a name on itself

I think a dimension actually always has a name, but doesn't necessarily have coordinates (and indeed coordinates are not always linked to a dimension). If you create a DataArray from a numpy array without specifying any dims/coords, you get "dim_0" etc for the dimension names (without coordinates).